Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -495,6 +496,16 @@ private static string FindCommandToRun(FileInfoBase[] files, out IScriptHost scr
                         }
                     }
                 }
+
+                foreach (var supportedFileName in scriptHost.SupportedFileNames)
```

Comment:
That was an issue before, but with latest commit, SupportedFileNames would still end up winning over supported extensions. Yes, the logic is a bit tricky.